browser/resources/settings/brave_privacy_page/brave_tor_subpage.js

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import { html, PolymerElement } from 'chrome://resources/polymer/v3_0/polymer/polymer_bundled.min.js';
+import { WebUIListenerMixin } from 'chrome://resources/js/web_ui_listener_mixin.js';
+import { RouteObserverMixin } from '../router.js';
+import { PrefsMixin } from '../prefs/prefs_mixin.js';
+import { BraveTorBrowserProxyImpl } from './brave_tor_browser_proxy.js'
+import './brave_tor_bridges_dialog.js'
+
+const SettingBraveTorPageElementBase = RouteObserverMixin(WebUIListenerMixin(PrefsMixin(PolymerElement)))
+
+const Usage = Object.freeze({
+  NOT_USED: 0,
+  USE_BUILT_IN: 1,
+  USE_REQUESTED: 2,
+  USE_PROVIDED: 3,
+})
+
+/**
+ * 'settings-brave-tor-page' is the settings page containing
+ * brave's Tor settings.
+ */
+class SettingsBraveTorPageElement extends SettingBraveTorPageElementBase {
+  static get is() {
+    return 'settings-brave-tor-subpage'
+  }
+
+  static get template() {
+    return html`{__html_template__}`
+  }
+
+  static get properties() {
+    return {
+      loadedConfig_: Object,
+      
+      useBridges_: Number,
+
+      builtInBridgesTypes_: {
+        type: Array,
+        value: [
+          { name: "snowflake", value: 0 },
+          { name: "obfs4", value: 1 },
+          { name: "meek-azure", value: 2 }
+        ]
+      },
+
+      builtinBridges_: Number,
+      requestedBridges_: Array,
+      providedBridges_: Array,
+
+      isUsingBridges_: {
+        type: Boolean,
+        notify: true,
+        value: false
+      },
+
+      showRequestBridgesDialog_: Boolean,
+    }
+  }
+
+  browserProxy_ = BraveTorBrowserProxyImpl.getInstance()
+
+  ready() {
+    super.ready()
+    this.browserProxy_.getBridgesConfig().then((config) => {
+      this.loadedConfig_ = config
+      this.UpdateUseBridges_(config.use_bridges)
+      this.builtinBridges_ = config.use_builtin_bridges
+      this.requestedBridges_ = config.requested_bridges
+      this.providedBridges_ = config.provided_bridges
+
+      this.$.builtInBridgesType.value = this.builtinBridges_
+
+      switch (this.useBridges_) {
+        case Usage.NOT_USED:
+        case Usage.USE_BUILT_IN:
+          this.$.brigdesGroup.selected = 'useBuiltIn'
+          break
+        case Usage.USE_REQUESTED:
+          this.$.brigdesGroup.selected = 'useRequestedBridges'
+          break
+        case Usage.USE_PROVIDED:
+          this.$.brigdesGroup.selected = 'useProvidedBridges'
+          break
+      }
```

Comment:
Referring to DOM elements from the JS part of the component should be last result. The default pattern should be inverted where DOM attributes are bound to properties in the JS.

Looks like you should provide DOM-bindable properties for:
- the value for the brigdesGroup _(aside: spelling of ~brigdes~ bridges)_
- requestedBridges
- providedBridges

For example (but not limited to):
```html
<settings-radio-group selected={{useBridgesValue_}}
```
```js
// property
    useBridgesValue_: {
      type: String,
      computed: 'computeUseBridgesValue_(useBridges_)' // useBridgesValue_ observes useBridges_ changes
    },
// ...
// Function
computeUseBridgesValue_() {
    switch (this.useBridges_) {
       case Usage.NOT_USED:
       case Usage.USE_BUILT_IN:
            return 'useBuiltIn'
       case Usage.USE_REQUESTED:
            return 'useRequestedBridges'
       case Usage.USE_PROVIDED:
            return 'useProvidedBridges'
      }
}
```

_note that this example will be different if you want to do 2-way binding for the selected attribute. Then you won't even need to on-change handler._